src/renderer/components/settings/WindowZoom.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vesktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+import { Margins } from "@vencord/types/utils";
+import { Forms, Slider, useEffect, useState } from "@vencord/types/webpack/common";
+
+import { SettingsComponent } from "./Settings";
+
+export const WindowZoom: SettingsComponent = ({ settings }) => {
+    const [zoomFactor, setZoomFactor] = useState(settings.zoomFactor ?? 1);
+
+    useEffect(() => {
+        const handleZoomChange = event => {
+            console.log("zoom changed", event.detail);
+            setZoomFactor(event.detail);
+        };
+
+        window.addEventListener("zoomChanged", handleZoomChange);
```

Comment:
you shouldn't need this event. changing settings will already update your state & rerender the ui

Reply:
I removed the listener, and the UI element does not update when the zoom gets changed, even with a listener